tensorflow/lite/micro/micro_context.cc

```diff
@@ -35,11 +35,15 @@ void* MicroContext::AllocatePersistentBuffer(size_t bytes) {
 TfLiteStatus MicroContext::RequestScratchBufferInArena(size_t bytes,
                                                        int* buffer_idx) {
   return allocator_.RequestScratchBufferInArena(
-      bytes, graph_.GetCurrentSubgraphIndex(), buffer_idx);
+      graph_.GetAllocations(), bytes, graph_.GetCurrentSubgraphIndex(),
+      buffer_idx);
 }
 
 void* MicroContext::GetScratchBuffer(int buffer_idx) {
-  ScratchBufferHandle* handle = scratch_buffer_handles_ + buffer_idx;
+  ScratchBufferHandle* handle =
```

Comment:
nit: maybe add a todo with bug here:  I feel this logic should probably be abstracted into either micro_allocator or micro_graph because it is very related with the implementation of how scratch buffer is implemented and it is better to let micro_context just be a shim layer